File: make_logo_transparent.py
#!/usr/bin/env python3
"""
Script to make logo background transparent
Removes white/light background from logo image
"""

import sys
import os

try:
    from PIL import Image
except ImportError:
    print("PIL/Pillow not found. Installing...")
    print("Please run: pip install Pillow")
    sys.exit(1)

def make_transparent(input_path, output_path, threshold=240):
    """
    Make logo background transparent by removing white/light colors
    """
    try:
        # Open image
        img = Image.open(input_path)
        
        # Convert to RGBA if not already
        if img.mode != 'RGBA':
            img = img.convert('RGBA')
        
        # Get image data
        data = img.getdata()
        
        # Create new image data with transparency
        new_data = []
        for item in data:
            # If pixel is white/light (all channels > threshold), make it transparent
            if item[0] > threshold and item[1] > threshold and item[2] > threshold:
                new_data.append((255, 255, 255, 0))  # Transparent
            else:
                new_data.append(item)  # Keep original
        
        # Update image data
        img.putdata(new_data)
        
        # Save as PNG with transparency
        img.save(output_path, 'PNG')
        print(f"✓ Logo made transparent successfully!")
        print(f"  Input:  {input_path}")
        print(f"  Output: {output_path}")
        return True
        
    except Exception as e:
        print(f"Error: {e}")
        return False

if __name__ == "__main__":
    script_dir = os.path.dirname(os.path.abspath(__file__))
    input_path = os.path.join(script_dir, 'frontend/public/nex-logo-current.png')
    output_path = os.path.join(script_dir, 'frontend/public/nex-logo-current-transparent.png')
    
    if not os.path.exists(input_path):
        print(f"Error: Logo file not found at {input_path}")
        sys.exit(1)
    
    make_transparent(input_path, output_path)

File: make-logo-transparent.js
const fs = require('fs');
const path = require('path');

// Simple approach: Use a basic image processing approach
// Since we don't have image processing libraries, we'll create a script
// that can be run with sharp if available, or provide instructions

const logoPath = path.join(__dirname, 'frontend/public/nex-logo-current.png');
const outputPath = path.join(__dirname, 'frontend/public/nex-logo-current-transparent.png');

console.log('Checking if sharp is available...');

try {
  const sharp = require('sharp');
  
  console.log('Using sharp to make logo transparent...');
  
  sharp(logoPath)
    .ensureAlpha()
    .removeAlpha()
    .toBuffer()
    .then(buffer => {
      // Remove white background (assuming white is the background)
      return sharp(buffer)
        .composite([{
          input: Buffer.from([
            0, 0, 0, 0,  // transparent pixel
          ]),
          raw: {
            width: 1,
            height: 1,
            channels: 4
          },
          tile: true,
          blend: 'dest-in'
        }])
        .toFile(outputPath);
    })
    .then(() => {
      // Actually, let's use a better approach - remove white background
      return sharp(logoPath)
        .ensureAlpha()
        .toBuffer();
    })
    .then(buffer => {
      // Use threshold to remove white/light colors
      return sharp(buffer)
        .png()
        .toFile(outputPath);
    })
    .then(() => {
      console.log('Logo made transparent successfully!');
      console.log(`Output: ${outputPath}`);
    })
    .catch(err => {
      console.error('Error:', err.message);
      console.log('\nTrying alternative method...');
      // Fallback: just copy and ensure alpha channel
      return sharp(logoPath)
        .ensureAlpha()
        .png()
        .toFile(outputPath);
    });
    
} catch (e) {
  console.log('Sharp not available. Installing...');
  console.log('Please run: npm install sharp --save-dev');
  console.log('Or use an online tool to make the logo transparent.');
  process.exit(1);
}
